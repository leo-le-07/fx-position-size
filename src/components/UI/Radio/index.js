'use strict';
var __importDefault =
  (this && this.__importDefault) ||
  function(mod) {
    return mod && mod.__esModule ? mod : { default: mod };
  };
Object.defineProperty(exports, '__esModule', { value: true });
var react_1 = __importDefault(require('react'));
var UIRadio = function(_a) {
  var options = _a.options,
    selectedOption = _a.selectedOption,
    handleOptionChange = _a.handleOptionChange;
  return react_1.default.createElement(
    'div',
    null,
    options.map(function(option, index) {
      return react_1.default.createElement(
        'div',
        { className: 'radio', key: index },
        react_1.default.createElement(
          'label',
          null,
          react_1.default.createElement('input', {
            type: 'radio',
            value: option.value,
            checked: selectedOption === option.value,
            onChange: handleOptionChange,
          }),
          ' ',
          option.name
        )
      );
    })
  );
};
exports.default = UIRadio;
