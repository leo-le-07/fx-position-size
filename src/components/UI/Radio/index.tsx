import React from 'react';

interface IProps {
  options: Array<{ value: string; name: string }>;
  selectedOption: string;
  handleOptionChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

const UIRadio = ({ options, selectedOption, handleOptionChange }: IProps) => {
  return (
    <div>
      {options.map((option, index) => (
        <div className="radio" key={index}>
          <label>
            <input
              type="radio"
              value={option.value}
              checked={selectedOption === option.value}
              onChange={handleOptionChange}
            />{' '}
            {option.name}
          </label>
        </div>
      ))}
    </div>
  );
};

export default UIRadio;
