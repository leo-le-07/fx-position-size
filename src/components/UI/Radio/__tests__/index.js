import React from 'react';
import { mount } from 'enzyme';
import Radio from '../index.js';

describe('Radio', () => {
  test('should match snapshot', () => {
    const options = [
      { value: 'value-1', name: 'Value 1' },
      { value: 'value-2', name: 'Value 2' },
      { value: 'value-3', name: 'Value 3' },
    ];
    const wrapper = mount(
      <Radio
        options={options}
        selectedOption={'value-1'}
        handleOptionChange={() => {}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
