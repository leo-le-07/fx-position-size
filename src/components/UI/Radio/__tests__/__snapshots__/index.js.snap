// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should match snapshot 1`] = `
<Radio
  handleOptionChange={[Function]}
  options={
    Array [
      Object {
        "name": "Value 1",
        "value": "value-1",
      },
      Object {
        "name": "Value 2",
        "value": "value-2",
      },
      Object {
        "name": "Value 3",
        "value": "value-3",
      },
    ]
  }
  selectedOption="value-1"
>
  <div>
    <div
      className="radio"
      key="0"
    >
      <label>
        <input
          checked={true}
          onChange={[Function]}
          type="radio"
          value="value-1"
        />
         
        Value 1
      </label>
    </div>
    <div
      className="radio"
      key="1"
    >
      <label>
        <input
          checked={false}
          onChange={[Function]}
          type="radio"
          value="value-2"
        />
         
        Value 2
      </label>
    </div>
    <div
      className="radio"
      key="2"
    >
      <label>
        <input
          checked={false}
          onChange={[Function]}
          type="radio"
          value="value-3"
        />
         
        Value 3
      </label>
    </div>
  </div>
</Radio>
`;
